### PR TITLE
Hotfix/3004 messages never go away when navigating through content in the sidebar

### PR DESCRIPTION
**Pull request by MCGallaspy**
Replaces #3031.

### CODE REVIEW

**Comment by rtibbles** [comment]
Seems to be a genuine test failure: "NameError: global name 'WebDriverException' is not defined"

An error in the test, no less!

**Comment by MCGallaspy** [comment]
Oh? How did that one slip in there?

File: static-libraries/js/khan-lite.js [hunk 1]
```diff
@@ -251,6 +251,11 @@ function clear_messages(msg_type) {
     return $("#message_container");
 }
 
+// To clear all messages including django and ajax.
+function clearAllMessages(msg_type) {
```
Comment (rtibbles) [inline static-libraries/js/khan-lite.js]: What is the purpose of the argument here, and what does this function do that clear_messages does not? If messages are appearing outside of #message_container that is a problem that should be addressed, not the addition of an almost identical function to the one directly above here.

Comment (rtibbles) [inline static-libraries/js/khan-lite.js]: @djallado ^

**Comment by djallado** [comment]
@rtibbles. 
Sorry. Yah, the argument has no purpose at all. This clearAllMessages() should clear django message in #message_container.

File: static-libraries/js/khan-lite.js (hunk 1 above)
Comment (djallado) [inline static-libraries/js/khan-lite.js]: The argument has no purpose at all. The function should clear django messages or normal messages loaded in #message_container.

**Comment by MCGallaspy** [comment]
>If messages are appearing outside of #message_container that is a problem that should be addressed, not the addition of an almost identical function to the one directly above here.

That is indeed the issue here.

**Comment by MCGallaspy** [comment]
I mean messages appearing outside the #message_container.

**Comment by MCGallaspy** [comment]
@rtibbles I'm convinced the 2.6 tests are failing just because the testing server is too pokey right now. Further, the inability to automatically merge is because of some front-end changes from @Antrikshy that I just merged (#2946), which won't affect the tests. So if you (or someone else) signs off on this I'm going to merge it.

**Comment by rtibbles** [comment]
Our test suite seems to be broken as far as 2.6 is concerned. Merge it, and we'll suffer the consequences.

**Pull request merged**